graphql-cli/src/main/java/io/ballerina/graphql/generator/ballerina/ConfigGenerator.java

```diff
@@ -0,0 +1,309 @@
+    private static final String COMPRESSION_NEVER = "COMPRESSION_NEVER";
+    private static final String CIRCUIT_BREAKER_CONFIG = "CircuitBreakerConfig";
+    private static final String ROLLING_WINDOW = "RollingWindow";
+    private static final String RETRY_CONFIG = "RetryConfig";
+    private static final String RESPONSE_LIMIT_CONFIGS = "ResponseLimitConfigs";
+    private static final String CLIENT_SECURE_SOCKET = "ClientSecureSocket";
+    private static final String TRUST_STORE = "TrustStore";
+    private static final String KEY_STORE = "KeyStore";
+    private static final String CERT_KEY = "CertKey";
+    private static final String CERT_VALIDATION_TYPE = "CertValidationType";
+    private static final String PROTOCOL = "Protocol";
+    private static final String POOL_CONFIGURATION = "PoolConfiguration";
+    private static final String HTTP_VERSION = "HttpVersion";
+    private static final String BEARER_TOKEN_CONFIG = "BearerTokenConfig";
+    private static final String REFRESH_TOKEN_GRANT_CONFIG = "RefreshTokenGrantConfig";
+    private static final String CLIENT_CREDENTIALS_GRANT_CONFIG = "ClientCredentialsGrantConfig";
+    private static final String CREDENTIAL_BEARER = "CredentialBearer";
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigGenerator.class);
+
+    public static ConfigGenerator getInstance() {
+        if (configGenerator == null) {
+            configGenerator = new ConfigGenerator();
+        }
+        return configGenerator;
+    }
+
+    public SyntaxTree generateConfigRecord(AuthConfig authConfig) throws IOException {
+        Set<String> commonTypesList = new LinkedHashSet<>(Arrays.asList(CONNECTION_CONFIG, CLIENT_HTTP1_SETTINGS,
+                PROXY_CONFIG, KEEP_ALIVE, CHUNKING, KEEPALIVE_AUTO, KEEPALIVE_ALWAYS, KEEPALIVE_NEVER, CHUNKING_AUTO,
+                CHUNKING_ALWAYS, CHUNKING_NEVER, CLIENT_HTTP2_SETTINGS, CACHE_CONFIG, CACHING_POLICY,
+                CACHE_CONTROL_AND_VALIDATORS, RFC_7234, COMPRESSION, COMPRESSION_AUTO, COMPRESSION_ALWAYS,
+                COMPRESSION_NEVER, CIRCUIT_BREAKER_CONFIG, ROLLING_WINDOW, RETRY_CONFIG, RESPONSE_LIMIT_CONFIGS,
+                CLIENT_SECURE_SOCKET, TRUST_STORE, KEY_STORE, CERT_KEY, CERT_VALIDATION_TYPE, PROTOCOL,
+                POOL_CONFIGURATION, HTTP_VERSION));
+
+        Set<String> clientConfigTypesList = new LinkedHashSet<>();
+        if (authConfig.getAuthConfigTypes().contains(AuthConfigType.BEARER)) {
+            clientConfigTypesList.add(BEARER_TOKEN_CONFIG);
+        }
+        if (authConfig.getAuthConfigTypes().contains(AuthConfigType.OAUTH2_REFRESH_TOKEN)) {
+            clientConfigTypesList.addAll(Arrays.asList(REFRESH_TOKEN_GRANT_CONFIG, CREDENTIAL_BEARER));
+        }
+        if (authConfig.getAuthConfigTypes().contains(AuthConfigType.BASIC)) {
+            commonTypesList.addAll(Arrays.asList(CLIENT_CREDENTIALS_GRANT_CONFIG, CREDENTIAL_BEARER));
```

Comment:
Let's add unit tests for these scenarios

Reply:
Added in https://github.com/ballerina-platform/graphql-tools/commit/eb969edf9f95622fe8d6dc7ecc000346998b403e